dll/win32/shell32/wine/changenotify.c

```diff
@@ -333,12 +369,96 @@ static BOOL should_notify( LPCITEMIDLIST changed, LPCITEMIDLIST watched, BOOL su
         return TRUE;
     return FALSE;
 }
+#else   /* def __REACTOS__ */
+LPITEMIDLIST PidlFromPathW(LPCWSTR path)
+{
+#if 1 /* FIXME: This is a workaround to fix display icon. */
+    if (PathFileExistsW(path))
+        return ILCreateFromPathW(path);
+#endif
+    return SHSimpleIDListFromPathW(path);
+}
+#endif  /* def __REACTOS__ */
 
 /*************************************************************************
  * SHChangeNotify				[SHELL32.@]
  */
 void WINAPI SHChangeNotify(LONG wEventId, UINT uFlags, LPCVOID dwItem1, LPCVOID dwItem2)
 {
+#ifdef __REACTOS__
+    LPITEMIDLIST pidl1 = NULL, pidl2 = NULL, pidlTemp1 = NULL, pidlTemp2 = NULL;
+    DWORD dwTick = GetTickCount();
+    WCHAR szPath1[MAX_PATH], szPath2[MAX_PATH];
+    LPWSTR psz1, psz2;
+    TRACE("(0x%08x,0x%08x,%p,%p)\n", wEventId, uFlags, dwItem1, dwItem2);
+
+    switch (uFlags & SHCNF_TYPE)
+    {
+        case SHCNF_IDLIST:
+            if (wEventId == SHCNE_FREESPACE)
+            {
+                DoNotifyFreeSpace(dwItem1, dwItem2);
+                goto Quit;
+            }
+            pidl1 = (LPITEMIDLIST)dwItem1;
+            pidl2 = (LPITEMIDLIST)dwItem2;
+            break;
+
+        case SHCNF_PATHA:
+            psz1 = psz2 = NULL;
+            if (dwItem1)
+            {
+                SHAnsiToUnicode((LPCSTR)dwItem1, szPath1, ARRAYSIZE(szPath1));
```

Comment:
Cast is not needed here (and below) since this is C.

Reply:
Deleted.